CMake/OtherTests.cmake

```
@@ -142,6 +142,7 @@ elseif(NOT HAVE_GETADDRINFO)
   set(HAVE_GETADDRINFO_THREADSAFE FALSE)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "AIX" OR
        CMAKE_SYSTEM_NAME STREQUAL "Darwin" OR
+       CMAKE_SYSTEM_NAME STREQUAL "iOS" OR
        CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" OR
```

```suggestion
elseif(APPLE OR
       CMAKE_SYSTEM_NAME STREQUAL "AIX" OR
       CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" OR
```
Would there be any downside to use `APPLE`? It includes these two flavours and the remaining ones.

---

There is a similar case in `CMake/OtherTests.cmake`. This is the patch I have pending for it; we could include it in this PR, if correct:
```diff
--- a/CMake/OtherTests.cmake
+++ b/CMake/OtherTests.cmake
@@ -86,9 +86,7 @@ endif()
 unset(CMAKE_TRY_COMPILE_TARGET_TYPE)
.
 if(NOT CMAKE_CROSSCOMPILING)
-  if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "iOS")
-    # only try this on non-apple platforms
-
+  if(NOT APPLE)
     # if not cross-compilation...
     set(CMAKE_REQUIRED_FLAGS "")
     if(HAVE_SYS_POLL_H)
```

---

Thank you very much for the suggestion! Using APPLE is even a better solution and with that all the Apple platforms are covered: https://cmake.org/cmake/help/latest/variable/APPLE.html 

I updated my PR with your suggestion, and I also added change the other line you mentioned. I tested locally and  it worked properly on my side.

Can you please check it again?

---

LGTM!